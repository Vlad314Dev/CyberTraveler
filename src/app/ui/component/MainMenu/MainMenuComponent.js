import './MainMenuStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import MainMenuOptionComponent from 'UIComponent/MainMenuOption';
import MainMenuTitle from 'UIComponent/MainMenuTitle';
import {
    AUTHORIZE_OPTION,
    DEFAULT_OPTION,
    PLAY_GAME_OPTION,
    SETTINGS_OPTION,
    VIEW_LEADERBOARD_OPTION,
} from 'UIStore/MainMenu/MainMenuConfig';

class MainMenuComponent extends PureComponent
{
    static propTypes = {
        activeOption: PropTypes.string.isRequired
    };

    render()
    {
        const { 
            activeOption,
        } = this.props;

        if (activeOption === DEFAULT_OPTION) {
            return (
                <div block="MainMenu" elem="Wrapper">
                    <MainMenuTitle />
                    <MainMenuOptionComponent />
                </div>
            );
        }

        if (activeOption === AUTHORIZE_OPTION) {
            return null;
        }

        if (activeOption === SETTINGS_OPTION) {
            return null;
        }

        if (activeOption === VIEW_LEADERBOARD_OPTION) {
            return null;
        }

        // If you're here then something went wrong
        if (activeOption === PLAY_GAME_OPTION) {
            return null;
        }

        return null;
    }
}

export default MainMenuComponent;
