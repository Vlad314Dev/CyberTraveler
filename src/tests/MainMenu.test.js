/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import MainMenuComponent from '../app/ui/component/MainMenu/MainMenuComponent';

describe('MainMenu', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <MainMenuComponent 
                activeOption={ '' }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <MainMenuComponent 
                activeOption={ '' }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
