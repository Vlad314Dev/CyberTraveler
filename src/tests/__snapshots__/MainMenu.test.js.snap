// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu Should render layout 1`] = `
Array [
  <div
    className="MainMenu__Wrapper"
  >
    <Memo(Connect(MainMenuTitleContainer)) />
  </div>,
]
`;
